### PR TITLE
[Refactor] Refactor drop table

### PR DESCRIPTION
**Pull request by sduzh**
Add a new method `Table.delete()` and override this method
in `OlapTable` and `LakeTable` for handling specific table logics.

## What type of PR is this：
- [ ] bug
- [ ] feature
- [ ] enhancement
- [x] refactor
- [ ] others

## Which issues of this PR fixes ：

Fixes #

### CODE REVIEW

File: fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java [hunk 1]
```diff
@@ -4487,50 +4496,19 @@ public HashMap<Long, AgentBatchTask> onEraseOlapOrLakeTable(OlapTable olapTable,
...
-
-                        // drop shard and lake tablet
-                        if (olapTable.isLakeTable()) {
-                            tabletIds.add(tabletId);
-                        }
+        colocateTableIndex.removeTable(olapTable.getId());
 
-                    } // end for tablets
-                } // end for indices
-            } // end for partitions
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/server/LocalMetastore.java:4523]: Moved this code block to class `OlapTable.DeleteTableTask`

File: fe/fe-core/src/main/java/com/starrocks/lake/LakeTable.java [hunk 2]
```diff
@@ -110,7 +116,58 @@ public void write(DataOutput out) throws IOException {
...
+        private final LakeTable table;
+        private final boolean replay;
+
+        DeleteTableTask(LakeTable table, boolean replay) {
+            this.table = table;
+            this.replay = replay;
+        }
+
+        @Override
+        public void run() {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/lake/LakeTable.java:143]: This method was extracted from the original method `LocalMetaStore.onEraseOlapOrLakeTable()`

File: fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java [hunk 3]
```diff
@@ -1781,14 +1787,80 @@ public void onCreate() {
...
+
+    private static class DeleteTableTask implements Runnable {
+        private final OlapTable table;
+
+        public DeleteTableTask(OlapTable table) {
+            this.table = table;
+        }
+
+        @Override
+        public void run() {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/catalog/OlapTable.java:1815]: This method was extracted from the original method `LocalMetaStore.onEraseOlapOrLakeTable()`

File: fe/fe-core/src/main/java/com/starrocks/catalog/Database.java [hunk 4]
```diff
@@ -355,104 +351,61 @@ public boolean createTable(Table table) {
...
+        } else {
+            runnable = table.delete(getId(), isReplay);
         }
 
-        LOG.info("finished dropping table[{}] in db[{}], tableId: {}", table.getName(), getOriginName(),
-                table.getId());
-        return batchTaskMap;
-    }
-
-    public void sendDropTabletTasks(HashMap<Long, AgentBatchTask> batchTaskMap) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:432]: Moved to `OlapTable.DeleteTableTask.run()`

File: fe/fe-core/src/main/java/com/starrocks/catalog/Database.java [hunk 5]
```diff
@@ -355,104 +351,61 @@ public boolean createTable(Table table) {
...
-                batchTaskMap = GlobalStateMgr.getCurrentState().onEraseOlapOrLakeTable((OlapTable) oldTable, isReplay);
-            }
-        } else { 
-            if (table.isOlapOrLakeTable()) {
-                batchTaskMap = GlobalStateMgr.getCurrentState().onEraseOlapOrLakeTable((OlapTable) table, isReplay);
-            }
-        }
-
-        // process related materialized views
-        if (table.isOlapTable()) {
```
Comment (sduzh) [inline fe/fe-core/src/main/java/com/starrocks/catalog/Database.java:416]: Moved to `LocalMetaStore.onEraseTable()`

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 80 / 141 (56.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|38|02.63%|[126, 127, 128, 129, 130, 131, 133, 134, 138, 139, 146, 147, 148, 149, 154, 155, 156, 157, 158, 159, 160, 161, 162, 164, 165, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|2|4|50.00%|[742, 743]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|41|55|74.55%|[1852, 1853, 1854, 1855, 1856, 1857, 1858, 1859, 1860, 1861, 1863, 1864, 1865, 1867]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|16|21|76.19%|[362, 363, 366, 367, 392]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|18|21|85.71%|[411, 455, 1815]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|[]|

**Comment by sduzh** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]

:disappointed: **fail** : 80 / 141 (56.74%)




#### file detail

|   |path|covered_line|new_line|coverage|not_covered_line_detail|
|----|----|----|----|----|----|
|:large_blue_circle:|com/starrocks/lake/LakeTable.java|1|38|02.63%|[126, 127, 128, 129, 130, 131, 133, 134, 138, 139, 146, 147, 148, 149, 154, 155, 156, 157, 158, 159, 160, 161, 162, 164, 165, 169, 170, 171, 172, 173, 174, 175, 176, 177, 178, 179, 180]|
|:large_blue_circle:|com/starrocks/catalog/CatalogRecycleBin.java|2|4|50.00%|[742, 743]|
|:large_blue_circle:|com/starrocks/catalog/OlapTable.java|41|55|74.55%|[1852, 1853, 1854, 1855, 1856, 1857, 1858, 1859, 1860, 1861, 1863, 1864, 1865, 1867]|
|:large_blue_circle:|com/starrocks/catalog/Database.java|16|21|76.19%|[360, 361, 364, 365, 389]|
|:large_blue_circle:|com/starrocks/server/LocalMetastore.java|18|21|85.71%|[411, 455, 1812]|
|:large_blue_circle:|com/starrocks/catalog/Table.java|2|2|100.00%|[]|

**Comment by sduzh** [comment]
run starrocks_admit_test

**Review by gengjun-git — APPROVED** [review]

**Comment by sduzh** [comment]
run starrocks_admit_test

**Review by imay — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**